code/modules/spells/roguetown/acolyte/dendor.dm

```diff
@@ -153,3 +153,32 @@
 		return TRUE
 	revert_cast()
 	return FALSE
+
+/obj/effect/proc_holder/spell/invoked/goodberry
+	name = "Nature's Favor"
+	desc = "Conjures blessed food from wild bushes. Yield may increase with holy skill."
+	overlay_state = "blesscrop"
+	releasedrain = 15
+	chargedrain = 0
+	chargetime = 5 SECONDS
+	range = 1
+	sound = 'sound/magic/churn.ogg'
+	invocations = list("May the wilds grant me sustenance.")
+	invocation_type = "whisper"
+	miracle = TRUE
+	associated_skill = /datum/skill/magic/holy
+	devotion_cost = 30
+	recharge_time = 30 MINUTES
+
+/obj/effect/proc_holder/spell/invoked/goodberry/cast(list/targets, mob/user = usr)
+	. = ..()
+	if(istype(targets[1], /obj/structure/flora/roguegrass/bush))
+		var/baseyield = rand(1,3)
+		var/skill = user.get_skill_level(/datum/skill/magic/holy)
+		var/fullyield = round(skill / 2) + baseyield
+		var/turf/t = get_turf(targets[1])
+		for(var/i in 1 to fullyield)
+			new /obj/item/reagent_containers/food/snacks/grown/berries/rogue/dendor(t)
+		return TRUE
+	revert_cast()
+	return FALSE
```

Comment:
With that kind of cooldown I'd suggest converting the yield of the bush itself to give the berries.